File: src/backend/Domain/Models/Location.cs
namespace Domain.Models;

public class Location
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = "shelf";
    public int Capacity { get; set; } = 12;
    public int? Humidity { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: src/frontend/web/src/types/location.ts
export type LocationType = 'shelf' | 'drybox'

export interface LocationResponse {
  id: string
  name: string
  type: LocationType
  capacity: number
  humidity: number | null
  createdAt: string
}

export interface AddLocationRequest {
  name: string
  type?: LocationType
  capacity?: number
  humidity?: number
}

export interface UpdateLocationRequest {
  name?: string
  type?: LocationType
  capacity?: number
  humidity?: number
}
